kombu/transport/consul.py

```diff
@@ -0,0 +1,272 @@
+            raise ChannelError('Cannot add key {0!r} to consul'.format(key))
+
+    def _get(self, queue, timeout=None):
+        """
+        Get the first available message from the queue
+
+        Before it does so it obtains a lock on the Key/Value store so
+        only one node reads at the same time. This is for read consistency
+        """
+        if self._obtain_lock(queue) is False:
+            raise Empty
+
+        key = '%s/msg/' % self._key_prefix(queue)
+        LOGGER.debug('Fetching key %s with index %s', key, self.index)
+        self.index, data = self.client.kv.get(key=key, recurse=True,
+                                              index=self.index,
+                                              wait=self.timeout)
+
+        try:
+            if data is None:
+                raise Empty
+
+            LOGGER.debug('Removing key %s with modifyindex %s',
+                         data[0]['Key'], data[0]['ModifyIndex'])
+
+            self.client.kv.delete(key=data[0]['Key'],
+                                  cas=data[0]['ModifyIndex'])
+
+            return loads(data[0]['Value'])
+        except TypeError:
+            pass
+        finally:
+            self._release_lock(queue)
+
+        raise Empty
+
+    def _purge(self, queue):
+        self._destroy_session(queue)
+        return self.client.kv.delete(key='%s/msg/' % self._key_prefix(queue),
+                                     recurse=True)
+
+    def _size(self, queue):
+        size = 0
+        try:
+            key = '%s/msg/' % self._key_prefix(queue)
+            LOGGER.debug('Fetching key recursively %s with index %s', key,
+                         self.index)
+            self.index, data = self.client.kv.get(key=key, recurse=True,
+                                                  index=self.index,
+                                                  wait=self.timeout)
+            size = len(data)
+        except TypeError:
+            pass
+
+        LOGGER.debug('Found %d keys under %s with index %s', size, key,
+                     self.index)
+        return size
+
+
+class Transport(virtual.Transport):
+    """
+    Consul K/V storage Transport for Kombu
+    """
+    Channel = Channel
+
+    default_port = DEFAULT_PORT
+    driver_type = 'consul'
+    driver_name = 'consul'
+
+    def __init__(self, *args, **kwargs):
+        if consul is None:
+            raise ImportError('Missing python-consul library')
+
+        super(Transport, self).__init__(*args, **kwargs)
+
+        self.connection_errors = (
+            virtual.Transport.connection_errors + (
+                consul.ConsulException, consul.base.ConsulException
+            )
+        )
+
+        self.channel_errors = (
+            virtual.Transport.channel_errors + (
+                consul.ConsulException, consul.base.ConsulException
+            )
+        )
+
+    def verify_connection(self, connection):
```

Comment:
Does this establish a new connection?

Most transports will not be able to check that a connection is still connected, so it's perfectly
reasonable to just return True here.